packages/file_selector/file_selector_windows/lib/src/file_selector_api.dart

```diff
@@ -0,0 +1,107 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/cupertino.dart';
+import 'package:win32/win32.dart';
+
+import 'file_selector_dart/dialog_mode.dart';
+import 'file_selector_dart/dialog_wrapper.dart';
+import 'file_selector_dart/dialog_wrapper_factory.dart';
+import 'file_selector_dart/selection_options.dart';
+
+/// File dialog handling for Open and Save operations.
+class FileSelectorApi {
+  /// Creates a new instance of [FileSelectorApi].
+  /// Allows Dependency Injection of a [DialogWrapperFactory] to handle dialog creation.
+  FileSelectorApi(this._dialogWrapperFactory)
+      : _foregroundWindow = GetForegroundWindow();
```

Comment:
Should we get the foreground window each time this API is called? Imagine the app creates multiple Flutter engines & windows, will this work as expected?

Reply:
Applied. Now, when the plugin is registered it calls `GetActiveWindow()` instead of `GetForegroundWindow()`.

Also, changing the used method solved an issue in the example application where after opening a file you weren't able to type text in the fields in the `SaveTextWidget` page.